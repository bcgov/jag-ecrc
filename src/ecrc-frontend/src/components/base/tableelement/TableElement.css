td {
  vertical-align: top;
  border: 5px solid white;
  padding: 3px;
  white-space: pre-wrap;
}
