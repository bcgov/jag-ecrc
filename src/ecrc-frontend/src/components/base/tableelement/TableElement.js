import React from "react";
import PropTypes from "prop-types";

import "./TableElement.css";

export default function TableElement({ element: { name, value } }) {
  return (
    <tr>
      <td>{name}</td>
      <td>{value}</td>
    </tr>
  );
}

TableElement.propTypes = {
  element: PropTypes.shape({
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
  }).isRequired
};
