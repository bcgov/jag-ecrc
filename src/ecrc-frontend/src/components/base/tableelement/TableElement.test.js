import React from "react";
import { create } from "react-test-renderer";

import TableElement from "./TableElement";

describe("TableElement Component", () => {
  test("Matches the snapshot", () => {
    const element = {
      name: "Somewhere",
      value: "Some value"
    };

    const singleElement = create(<TableElement element={element} />);
    expect(singleElement.toJSON()).toMatchSnapshot();
  });
});
