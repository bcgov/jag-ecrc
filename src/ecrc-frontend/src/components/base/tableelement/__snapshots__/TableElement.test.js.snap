// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableElement Component Matches the snapshot 1`] = `
<tr>
  <td>
    Somewhere
  </td>
  <td>
    Some value
  </td>
</tr>
`;
